src/js/components/ImageMenu.vue

```diff
@@ -0,0 +1,219 @@
+            align="middle">
+            <el-col :span="2">
+              <span class="drag-handle image-select-entry" style="float: left">☰</span>
+            </el-col>
+            <el-col :span="8">
+              <span class="image-select-entry">
+                &nbsp;{{corpus.images.get(image).type | formatImageType}}
+              </span>
+            </el-col>
+            <el-col :span="10">
+              <input
+              class="image-select-entry"
+              type="range"
+              min="0"
+              max="1.0"
+              step="0.001"
+              @input="setOpacity(image, $event.target.value)"/>
+            </el-col>
+            <el-col :span="4">
+              <span>
+                <i class="fa fa-eye image-select-entry"
+                  :style="{color: imageSettings[image].visible ? 'green' : 'red'}"
+                  @click="toggleVisible(image)">
+                </i>
+              </span>
+            </el-col>
+          </el-row>
+        </el-option>
+      </el-select>
+    </el-col>
+    <el-col :span="1">
+      <span class="label">Zoom</span>
+    </el-col>
+    <el-col :span="4">
+      <el-slider
+        class="image-zoom"
+        v-model="changeZoom"
+        :min="0.1"
+        :step="0.01"
+        :max="1.0"
+        :format-tooltip="formatTooltip">
+      </el-slider>
+    </el-col>
+    <el-col v-if="roiEditable" v-show="artefact && artefact !== 'new'"  :span="4">
+      <el-radio-group v-model="changeViewMode" size="mini">
+        <el-radio-button label="ROI">{{$i18n.str('ROI')}}</el-radio-button>
+        <el-radio-button label="ART">{{$i18n.str('ART')}}</el-radio-button>
+      </el-radio-group>
+    </el-col>
+    <el-col v-show="artefact || artefact === 'new'"  :span="3">
+      <el-button @click="toggleMask" size="mini">Mask</el-button>
+    </el-col>
+    <el-col v-if="roiEditable" v-show="viewMode === 'ROI' && artefact" :span="3">
+      <el-button @click="delSelectedRoi" size="mini">Del ROI</el-button>
+    </el-col>
+    <el-col v-if="artefactEditable" v-show="viewMode === 'ART' && (artefact || artefact === 'new')" :span="3">
+      <el-button
+              @click="toggleDrawingMode"
+              :type="drawingMode === 'draw' ? 'primary' : 'warning'"
+              size="mini">
+        {{drawingMode === 'draw' ? 'Draw' : 'Erase'}}
+      </el-button>
+    </el-col>
+    <el-col v-if="artefactEditable" v-show="viewMode === 'ART' && (artefact || artefact === 'new')" :span="4">
+      <el-slider
+        v-model="changeBrushSize"
+        :min="0"
+        :max="200"
+        :step="1">
+      </el-slider>
+    </el-col>
+  </el-row>
+</template>
+
+<script>
+/**
+ * This component has a lot of emit functions.  Perhaps it will be better
+ * to create a modular container that holds this menu and the possible
+ * image view modules that can accompany it.  If we want the menu to be
+ * a component, however, I see no way around using these emit functions.
```

Comment:
That sounds cool.  I wonder if Vue won't eventually add support for two-way binding, since the pattern you mention is rather close to that.  My sense is that your way will be a cleaner, more readable, implementation, but perhaps goes a bit against the "original intent" of the designers ;).  Maybe I try it out in another iteration, thanks.